eras/conway/impl/src/Cardano/Ledger/Conway/Delegation/Certificates.hs

```diff
@@ -89,48 +130,59 @@ instance NFData (ConwayDelegCert c)
 
 instance NoThunks (ConwayDelegCert c)
 
-data ConwayDCert c
-  = ConwayDCertDeleg !(ConwayDelegCert c)
-  | ConwayDCertPool !(PoolCert c)
-  | ConwayDCertConstitutional !(ConstitutionalDelegCert c)
+data ConwayDCert era
+  = ConwayDCertRegKey !(StakeCredential (EraCrypto era))
```

Comment:
let's remove this in favor of a new `ConwayRegKey` constructor in `ConwayDelegCert`.

```
ConwayRegKey !(StakeCredential c) !Coin
```